File: mvn-advent-of-code/src/test/java/advent/twenty_sixteen/Day21ScrambledLettersAndHashTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day21ScrambledLettersAndHashTest {
    @Test
    public void checkReal() {
        DailyProblem<String, String> dp = new Day21ScrambledLettersAndHash("/twenty_sixteen/Day21-ScrambledLettersAndHash-input.txt", "abcdefgh", "fbgdceah");

        assertEquals("bgfacdeh", dp.getPart1Answer());
        assertEquals("bdgheacf", dp.getPart2Answer());
    }

    @Test
    public void checkExample1() {
        DailyProblem<String, String> dp = new Day21ScrambledLettersAndHash("/twenty_sixteen/Day21-ScrambledLettersAndHash-example#1.txt", "abcde", null);

        assertEquals("decab", dp.getPart1Answer());
        // assertEquals("", dp.getPart2Answer());  -- Cannot uniquely decode a 5 character password
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day21ScrambledLettersAndHashTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day21ScrambledLettersAndHashTest  extends FlatSpec with Matchers {
  "2016 Day 21 - Input" should "calculate answers" in {
    val dp = new Day21ScrambledLettersAndHash("/twenty_sixteen/Day21-ScrambledLettersAndHash-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day21ScrambledLettersAndHash("/twenty_sixteen/Day21-ScrambledLettersAndHash-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
